lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -59,7 +60,23 @@ export class ClientCredentialClient extends BaseClient {
 
         const cachedAuthenticationResult =
             await this.getCachedAuthenticationResult(request);
+
         if (cachedAuthenticationResult) {
+            // if the token is not expired but must be refreshed; get a new one in the background
+            if (this.lastCacheOutcome === CacheOutcome.REFRESH_CACHED_ACCESS_TOKEN) {
+                this.logger.info(
+                    "ClientCredentialClient:getCachedAuthenticationResult - Cached access token's refreshOn property has been exceeded'. It's not expired, but must be refreshed."
+                );
+
+                // return the cached token, don't wait for the result of this request
+                const refreshAccessToken = true;
+                this.executeTokenRequest(request, this.authority, refreshAccessToken);
+            }
+
+            // reset the last cache outcome
+            this.lastCacheOutcome = CacheOutcome.NO_CACHE_HIT;
```

Comment:
Isn't `lastCacheOutcome` used elsewhere? In case of "refresh_in" you are resetting it to "NO_CACHE_HIT" ?

Reply:
It's not used elsewhere. We set it in the function that attempts to get the token from the cache. We're just resetting the value here for the next time acquire token is called.